src/pkg/docs/service/processing/processing.go

```diff
@@ -184,6 +187,45 @@ func (p *Proc) GetRetrieveStatus(CID *cid.Cid) (Status, error) {
 	return ret.Status, nil
 }
 
+type Message struct {
+	ReqID  string
+	Status Status
+}
+
+type subscriber struct {
+	active bool
+	notify *func(sub Message)
+}
+
+func (s *subscriber) Disable() {
+	s.active = false
+}
+
+func (s *subscriber) isActive() bool {
+	return s.active
+}
+
+func (s *subscriber) Notify(msg interface{}) {
+	if !s.isActive() {
+		return
+	}
+
+	(*s.notify)(msg.(Message))
+}
+
+func (p *Proc) Subscribe(f func(sub Message)) *subscriber {
+	sub := subscriber{
+		active: true,
+		notify: &f,
+	}
+	p.publisher.AddSubscriber() <- &sub
```

Comment:
Pass the pointer into the channel it's bad practice. 
And here we need to use the variable name `msg` instead of `sub`.